src/PIL/GifImagePlugin.py

```diff
@@ -259,9 +258,10 @@ def _seek(self, frame):
                 # do not dispose or none specified
                 self.dispose = None
             elif self.disposal_method == 2:
-                # replace with background colour
-                self.dispose = Image.core.fill("P", self.size,
-                                               self.info["background"])
+                # replace with transparency if there is one, otherwise
+                # background colour
```

Comment:
Why potentially use the transparency? I don't see this in the spec.

Reply:
I've found https://legacy.imagemagick.org/Usage/anim_basics/#dispose
> is cleared to transparency
> ...
> There is some thinking that rather than clearing the overlaid area to the transparent color, this disposal should clear it to the 'background' color meta-data setting stored in the GIF animation
> ...
> However all modern web browsers clear just the area that was last overlaid to transparency, as such this is now accepted practice, and what IM now follows.